agent/lib/kubernetes.js

```diff
@@ -238,7 +239,22 @@ Watcher.prototype.watch = function () {
             }
         });
     });
-    request.on('error', function(e) { log.error('error on watch: %s', e); });
+    request.on('error', function(e) {
+        log.error('error on watch: %s', e);
+        if (!self.closed) {
```

Comment:
The error event will generally be immediately followed by an end event which will immediately trigger the list. This code will also schedule a delayed list in that case, which to me at least is confusing.

Reply:
We have seen it to be the case that end might not be called, as there are no new watches set (see https://github.com/EnMasseProject/enmasse/issues/3542). This is intended to catch that case.

Reply:
Understood, but your code will run even if end *is* called, resulting in a bifurcation of sorts. If you have a timeout on the request, then whether end is called or not, the request can be aborted.

Reply:
Good point, maybe leave this out  for now and rely on the client timeout for now then.